extensions/GOOGLE/SPV_GOOGLE_hlsl_functionality1.asciidoc

```diff
@@ -0,0 +1,159 @@
+SPV_GOOGLE_hlsl_functionality1
+==============================
+
+Name Strings
+------------
+
+SPV_GOOGLE_hlsl_functionality1
+
+Contact
+-------
+
+See *Issues* list in the Khronos SPIRV-Headers repository:
+https://github.com/KhronosGroup/SPIRV-Headers
+
+Contributors
+------------
+
+- Hai Nguyen, Google
+- David Neto, Google
+- John Kessenich, Google
+- Lei Zhang, Google
+
+Status
+------
+- Draft
+
+Version
+-------
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | 2018-02-16
+| Initial            | 2
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification,
+Version 1.2 Revision 4, Unified.
+
+This extension requires SPIR-V 1.2.
+
+Overview
+--------
+
+This extension provides two new decorations to extend HLSL functionality:
+HlslCounterBuffer and HlslSemantic.
+
+The decoration HlslCounterBuffer is used with OpDecorateId to link a counter
+buffer to a UAV resources that has an associated counter.
+
+The decoration HlslSemantic is used with OpDecorateStringGOOGLE and
+OpMemberDeocrateStringGOOGLE to decorate an input or output variable id
+with a string representing semantic as defined in the HLSL source. The
+decoration HlslSemantic can also be used with future string decorations that
+may get added to SPIR-V.
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following
+*OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_GOOGLE_hlsl_functionality1"
+----
+
+New Capabilities
+----------------
+
+None.
+
+New Decorations
+---------------
+
+----
+HlslCounterBufferGOOGLE
+HlslSemanticGOOGLE
+----
+
+New Builtins
+------------
+None.
+
+New Instructions
+----------------
+None.
+
+Token Number Assignments
+------------------------
+
+[width="40%"]
+[cols="70%,30%"]
+[grid="rows"]
+|====
+|HlslCounterBufferGOOGLE | 5634
+|HlslSemanticGOOGLE      | 5635
+|====
+
+Modifications to the SPIR-V Specification, Version 1.0
+------------------------------------------------------
+(Modify Section 3.20, Decoration) ::
+
+[cols="1,10,5,2,2",options="header",width = "100%"]
+|====
+2+^.^| Decoration | Enabling Capabilities 2+<.^| Extra Operands
+| 5634 | *HlslCounterBufferGOOGLE* +
+An <id> of the counter buffer associated with the target buffer.
+  |
+  2+| _<id>_ +
+      Counter Buffer ID
+| 5635 | *HlslSemanticGOOGLE* +
+A string describing the intended use of a value.
+Use this to express user-defined and system semantic values for shaders
+originally written in HLSL.
```

Comment:
Say anything about case sensitivity? Or what case things should come out?

Reply:
Added a sentence to say this is case insensitive, following HLSL tradition. Whether we convey the original semantic string or do normalization in the compilers are implementation specific; I feel we should not mandate that in the spec.